src/aspire/utils/misc.py

```diff
@@ -206,33 +217,39 @@ def gaussian_3d(size, mu=(0, 0, 0), sigma=(1, 1, 1), indexing="zyx", dtype=np.fl
     in a 3D numpy array.
 
     :param size: The length of each dimension of the returned array (pixels)
-    :param mu: A 3-tuple, :math:`(\\mu_x, \\mu_y, \\mu_z)`, indicating the center of the Gaussian
-    :param sigma: A 3-tuple, :math:`(\\sigma_x, \\sigma_y, \\sigma_z)`, of the standard deviation
+    :param mu: Iterable of len(3), :math:`(\\mu_x, \\mu_y, \\mu_z)`, indicating the center of the Gaussian
+    :param sigma: Iterable of len(3) or constant, :math:`(\\sigma_x, \\sigma_y, \\sigma_z)`, of the standard deviation
             in the x, y, and z directions. A single value, :math:`\\sigma`, can be
             used when :math:`\\sigma_x = \\sigma_y = \\sigma_z`
+    :param indexing: The order of axis indexing, passed to grid_3d
     :param dtype: dtype of returned array
     :return: Numpy array (3D)
     """
     if np.ndim(sigma) == 0:
         sigma = (sigma, sigma, sigma)
-    else:
-        assert (
-            isinstance(sigma, tuple) and len(sigma) == 3
-        ), "sigma must be a scalar or 3-tuple."
+
+    mu = np.array(mu, dtype=dtype)
+    sigma = np.array(sigma, dtype=dtype)
+    if len(mu) != 3:
+        raise ValueError("`mu` must be len(3).")
+    if len(sigma) != 3:
```

Comment:
I would just throw in some tests to cover these checks, e.g. a length n-1 mu and a length n+1 sigma

Reply:
Good call. Done.